react/week3/todo-list-form/src/Todo.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import Border from "./Border";
+import PropTypes from "prop-types";
+
+const Todo = ({ todoItem, deleteTodo, updateTodo }) => {
+  const [status, setStatus] = useState(false);
+  const [isEdit, setIsEdit] = useState(false);
+  const [newDescription, setNewDescription] = useState(todoItem.description);
+
+  const checkStatus = () => {
+    return setStatus(!status);
+  };
+  const checkEdit = () => {
+    return setIsEdit(!isEdit);
+  };
+  console.log(isEdit);
+  const buttonUpdate = () => {
+    if (newDescription === "") {
+      alert("Enter a valid task");
+    } else {
+      updateTodo(todoItem.id, newDescription);
+      setIsEdit(!isEdit);
+    }
+  };
+
+  const editTemplate = (
+    <div>
+      <Border>
+        <input
+          className="edit-Input"
+          value={newDescription}
+          type="text"
+          onChange={(event) => {
+            setNewDescription(event.target.value);
+          }}
+        />
+        <button onClick={buttonUpdate} className="btnUpdate">
+          Update
+        </button>
+        <button className="btnDelete" onClick={() => deleteTodo(todoItem.id)}>
+          Delete
+        </button>
+      </Border>
+    </div>
+  );
+
+  const viewTemplate = (
+    <Border>
+      <span
+        className="todo"
+        style={{ textDecoration: status ? "line-through" : "none" }}
+      >
+        {todoItem.description} | {todoItem.deadline}
+      </span>
+
+      <input type="checkbox" value={todoItem.id} onClick={checkStatus}></input>
+
+      <button className="btnDelete" onClick={() => deleteTodo(todoItem.id)}>
+        Delete
+      </button>
+
+      <button className="btnEdit" onClick={checkEdit}>
+        Edit
+      </button>
+    </Border>
+  );
+  return <div>{isEdit ? editTemplate : viewTemplate}</div>;
```

Comment:
This definitely works but as good practice we try to avoid as much duplicate code writing as possible. As you have a lot of the same elements in both I can recommend doing the checking inside the "return" ()

E.g.: 
`      <button className="btnEdit" onClick={checkEdit}>
        {isEdit ? "Update" : "Edit"} // Which is short for if (isEdit) show "Update", otherwise show "Edit"
      </button>
`